### ISSUE TITLE
Smooth Quantile Loss

### ISSUE THREAD

**Issue by avorozhtsov** [catboost/catboost#1909]
Problem: Huber loss is smooth variant of L1. Smooth variants of all Quantile losses are needed. 

pytorch version:
```
class SmoothQLoss(nn.L1Loss):
    __constants__ = ['reduction', 'beta', 'alpha']

    def __str__(self):
        return f"SmoothQLoss(beta={self.beta}, alpha={self.alpha})"

    def __init__(self, alpha: float = 0.5, beta: float = 1.0):
        super().__init__()
        self.beta = beta
        self.alpha = alpha

    def forward(self, predict: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
        m = 2.0 * self.alpha - 1.0
        shift = self.beta * m

        return (
            F.smooth_l1_loss(target - shift, predict, reduction=self.reduction, beta=self.beta)
            + m * torch.mean(target - predict - 0.5 * shift)
        )
```
